adapters/conversant/conversant.go

```diff
@@ -1,291 +1,177 @@
 package conversant
 
 import (
-	"bytes"
-	"context"
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
-	"net/http"
-
 	"github.com/mxmCherry/openrtb"
 	"github.com/prebid/prebid-server/adapters"
 	"github.com/prebid/prebid-server/errortypes"
-	"github.com/prebid/prebid-server/pbs"
-	"golang.org/x/net/context/ctxhttp"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
 )
 
 type ConversantAdapter struct {
 	http *adapters.HTTPAdapter
 	URI  string
 }
 
-// Corresponds to the bidder name in cookies and requests
-func (a *ConversantAdapter) Name() string {
-	return "conversant"
-}
-
-// Return true so no request will be sent unless user has been sync'ed.
-func (a *ConversantAdapter) SkipNoCookies() bool {
-	return true
-}
-
-type conversantParams struct {
-	SiteID      string   `json:"site_id"`
-	Secure      *int8    `json:"secure"`
-	TagID       string   `json:"tag_id"`
-	Position    *int8    `json:"position"`
-	BidFloor    float64  `json:"bidfloor"`
-	Mobile      *int8    `json:"mobile"`
-	MIMEs       []string `json:"mimes"`
-	API         []int8   `json:"api"`
-	Protocols   []int8   `json:"protocols"`
-	MaxDuration *int64   `json:"maxduration"`
-}
-
-func (a *ConversantAdapter) Call(ctx context.Context, req *pbs.PBSRequest, bidder *pbs.PBSBidder) (pbs.PBSBidSlice, error) {
-	mediaTypes := []pbs.MediaType{pbs.MEDIA_TYPE_BANNER, pbs.MEDIA_TYPE_VIDEO}
-	cnvrReq, err := adapters.MakeOpenRTBGeneric(req, bidder, a.Name(), mediaTypes)
-
-	if err != nil {
-		return nil, err
+func (c ConversantAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+	if len(request.Imp) == 0 {
```

Comment:
Nice to have. Not strictly needed since PBS "Core" will not call `MakeRequests` if there are no imps. Consider removing the check, but I'm fine if you want to leave it in to be extra defensive.